api/internal/localizer/localizer_test.go

```diff
@@ -299,3 +299,171 @@ patches:
 	require.Error(t, lclzr.Localize())
 }
+
+func TestLocalizeDirInTarget(t *testing.T) {
+	type testCase struct {
+		name  string
+		files map[string]string
+	}
+	for _, tc := range []testCase{
+		{
+			name: "multi_nested_child",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta/epsilon
+kind: Kustomization
+`,
+				"delta/epsilon/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+namespace: kustomize-namespace
+`,
+			},
+		},
+		{
+			name: "recursive",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta
+kind: Kustomization
+`,
+				"delta/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+components:
+- epsilon
+kind: Component
+`,
+				"delta/epsilon/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+namespace: kustomize-namespace
+`,
+			},
+		},
+		{
+			name: "file_in_dir",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta
+kind: Kustomization
+`,
+				"delta/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+patches:
+- path: patch.yaml
+`,
+				"delta/patch.yaml": podConfiguration,
+			},
+		},
+		{
+			name: "multiple_calls",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta
+- delta/epsilon
+kind: Kustomization
+`,
+				"delta/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+namespace: kustomize-namespace
+`,
+				"delta/epsilon/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1alpha1
+buildMetadata:
+- managedByLabel
+kind: Component
+`,
+			},
+		},
+		{
+			name: "localize_directory_name_when_absent",
+			files: map[string]string{
+				"kustomization.yaml": fmt.Sprintf(`apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- %s
+kind: Kustomization
+`, LocalizeDir),
+				fmt.Sprintf("%s/kustomization.yaml", LocalizeDir): `apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
+namespace: kustomize-namespace
+`,
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			fSys := makeMemoryFs(t)
+			addFiles(t, fSys, "/alpha/beta/gamma", tc.files)
+
+			lclzr := createLocalizer(t, fSys, "/alpha/beta/gamma", "/alpha/beta", "/dst")
+			err := lclzr.Localize()
+			require.NoError(t, err)
+
+			fSysExpected := makeMemoryFs(t)
+			addFiles(t, fSysExpected, "/alpha/beta/gamma", tc.files)
+			addFiles(t, fSysExpected, "/dst/gamma", tc.files)
+			checkFSys(t, fSysExpected, fSys)
+		})
+	}
+}
+
+func TestLocalizeDirCleanedSibling(t *testing.T) {
+	fSys := makeMemoryFs(t)
+	kustAndComponents := map[string]string{
+		"beta/gamma/kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+components:
+- delta/../../../../a/b/../../alpha/beta/sibling
```

Comment:
Is the point that this is expressed in a ridiculous way that looks like it is trying to escape, but once we clean it, it turns out to be a legal path, so it works?